proto/config/netinst.proto

```diff
@@ -85,6 +85,25 @@ message IPRoute {
   string gateway = 2;
 }
 
+// DefaultRouteSelection defines how local network instance picks the default route
+// when it has multiple ports assigned (using adapter labels).
+enum DefaultRouteSelection {
+  // User did not specify how to select the default route for network instance.
+  // EVE will use the default option, which is connectivity testing for "uplink"
+  // and "freeuplink" port labels, and cost-based selection otherwise.
+  DEFAULT_ROUTE_SELECTION_UNSPECIFIED = 0;
+  // Periodically run connectivity tests against the port gateway IP (if available)
+  // and the controller URL (if management port), to apply default route from
+  // a port with working connectivity.
+  // This selection algorithm thus provides automatic fail-over between ports.
+  DEFAULT_ROUTE_SELECTION_CONNECTIVITY_TESTING = 1;
+  // Apply default route of the port with the lowest cost.
+  // This is limited to management ports because app-shared ports do not have cost defined.
```

Comment:
I'm not sure. Either we add cost to app-shared ports or this will be renamed to `DEFAULT_ROUTE_SELECTION_USER` and there will be another field in `NetworkInstanceConfig` to select the port to use by default (by name or label). The latter requires that the controller supports run-time change of the network instance config (which EVE does support).